In [ ]:
import os
import carbon_tracker

# Create the emissions folder if it doesn't exist
if not os.path.exists('./emissions'):
	os.makedirs('./emissions')

#improt file in "Sad-Brain-Analytics-Hackathon\carbon_tracker.py"
%run carbon_tracker.py
carbon_tracker.start_tracker()

# **Import libraries:**

In [1]:
import pandas as pd 
import matplotlib.pyplot as plt 
import seaborn as sns 

# **Prepare Data:**

In [2]:
df = pd.read_csv("final_depression_dataset_1.csv")
df.head()

Name  Gender  Age       City Working Professional or Student  \
0    Pooja  Female   37  Ghaziabad            Working Professional   
1  Reyansh    Male   60     Kalyan            Working Professional   
2    Manvi  Female   42     Bhopal            Working Professional   
3     Isha  Female   44      Thane            Working Professional   
4    Aarav    Male   48     Indore            Working Professional   

          Profession  Academic Pressure  Work Pressure  CGPA  \
0            Teacher                NaN            2.0   NaN   
1  Financial Analyst                NaN            4.0   NaN   
2            Teacher                NaN            2.0   NaN   
3            Teacher                NaN            3.0   NaN   
4     UX/UI Designer                NaN            4.0   NaN   

   Study Satisfaction  Job Satisfaction Sleep Duration Dietary Habits Degree  \
0                 NaN               4.0      7-8 hours       Moderate     MA   
1                 NaN               3.0      5-6 hours      Unhealthy  B.Com   
2                 NaN               3.0      5-6 hours       Moderate  M.Com   
3                 NaN               5.0      7-8 hours        Healthy     MD   
4                 NaN               3.0      7-8 hours       Moderate     BE   

  Have you ever had suicidal thoughts ?  Work/Study Hours  Financial Stress  \
0                                    No                 6                 2   
1                                   Yes                 0                 4   
2                                    No                 0                 2   
3                                   Yes                 1                 2   
4                                   Yes                 6                 5   

  Family History of Mental Illness Depression  
0                               No         No  
1                              Yes         No  
2                               No         No  
3                              Yes         No  
4                              Yes         No

In [6]:
df.shape

(2556, 19)

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2556 entries, 0 to 2555
Data columns (total 19 columns):
 #   Column                                 Non-Null Count  Dtype  
---  ------                                 --------------  -----  
 0   Name                                   2556 non-null   object 
 1   Gender                                 2556 non-null   object 
 2   Age                                    2556 non-null   int64  
 3   City                                   2556 non-null   object 
 4   Working Professional or Student        2556 non-null   object 
 5   Profession                             1883 non-null   object 
 6   Academic Pressure                      502 non-null    float64
 7   Work Pressure                          2054 non-null   float64
 8   CGPA                                   502 non-null    float64
 9   Study Satisfaction                     502 non-null    float64
 10  Job Satisfaction                       2054 non-null   float64
 11  Slee

### *Handling null values:*

In [7]:
df.isnull().sum()

Name                                        0
Gender                                      0
Age                                         0
City                                        0
Working Professional or Student             0
Profession                                673
Academic Pressure                        2054
Work Pressure                             502
CGPA                                     2054
Study Satisfaction                       2054
Job Satisfaction                          502
Sleep Duration                              0
Dietary Habits                              0
Degree                                      0
Have you ever had suicidal thoughts ?       0
Work/Study Hours                            0
Financial Stress                            0
Family History of Mental Illness            0
Depression                                  0
dtype: int64

Out of 2556 entries, the columns Academic Pressure, CGPA, and Study Satisfaction have 2054 null values each. Since these columns contain mostly missing data, removing them will not significantly impact our model training. On the other hand, Profession and Job Satisfaction are important features for our model and should be retained.

In [9]:
# Drop unecessary columns

df = df.drop(['Academic Pressure', 'CGPA', 'Study Satisfaction'], axis=1)
df.shape

(2556, 16)

In [11]:
# Replace null values with most commun
df = df.apply(lambda column: column.fillna(column.mode()[0]) if column.isna().any() else column)

In [12]:
df.isnull().sum()

Name                                     0
Gender                                   0
Age                                      0
City                                     0
Working Professional or Student          0
Profession                               0
Work Pressure                            0
Job Satisfaction                         0
Sleep Duration                           0
Dietary Habits                           0
Degree                                   0
Have you ever had suicidal thoughts ?    0
Work/Study Hours                         0
Financial Stress                         0
Family History of Mental Illness         0
Depression                               0
dtype: int64

 For memory efficiency, string dtypre ismore efficient than object because it is optimized for text data, while object dtype can store mixed types but uses more memory for purely textual data.